### PR TITLE
SqlProvider total count

### PR DESCRIPTION
**Pull request by SamMousa**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13883 

I'd like some feedback on the test.
- Does it have the right group
- Is it in the right folder
- Should it use sqlite (I assumed that's the most lightweight way of testing the database without mocking).

### CODE REVIEW

**Comment by SilverFire** [comment]
I think it should be done in a manner RBAC Manager tests: https://github.com/SamMousa/yii2/tree/master/tests/framework/rbac

**Review by SilverFire — APPROVED** [review]
Looks good

**Comment by SamMousa** [comment]
@SilverFire Isn't one db test enough considering the query itself is trivial? No use making tests slower?

**Comment by SilverFire** [comment]
Yeah, I think you are right as well

File: tests/framework/data/SqlDataProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public function testTotalCount()
+    {
+        $dataProvider = new SqlDataProvider([
+            'sql' => 'select * from `customer`',
+            'db' => $this->getConnection()
+        ]);
+        $this->assertEquals(3, $dataProvider->getTotalCount());
+    }
+
+
```
Comment (arogachev) [inline tests/framework/data/SqlDataProviderTest.php]: Unnecessary linebreaks.

**Review by arogachev — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

**Review by arogachev — APPROVED** [review]

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/79f16b149239c0d1314102999a2fc8353ee8cb0b. Thank you!

**Pull request closed without merging**